Formula/heartbeat.rb

```diff
@@ -0,0 +1,97 @@
+class Heartbeat < Formula
+  desc "Lightweight Shipper for Uptime Monitoring"
+  homepage "https://www.elastic.co/products/beats/heartbeat"
+  url "https://github.com/elastic/beats/archive/v5.6.4.tar.gz"
+  sha256 "c06f913af79bb54825483ba0ed4b31752db5784daf3717f53d83b6b12890c0a4"
+  head "https://github.com/elastic/beats.git"
+
+  depends_on "go" => :build
+
+  def install
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/elastic/beats").install Dir["{*,.git,.gitignore}"]
+
+    ENV["GOPATH"] = gopath
+
+    cd gopath/"src/github.com/elastic/beats/heartbeat" do
+      system "make"
+      libexec.install "heartbeat"
+
+      (etc/"heartbeat").install("heartbeat.yml")
+      (etc/"heartbeat").install("heartbeat.full.yml")
+      (etc/"heartbeat").install("heartbeat.template.json")
+      (etc/"heartbeat").install("heartbeat.template-es2x.json")
+      (etc/"heartbeat").install("heartbeat.template-es6x.json")
+      prefix.install_metafiles
+    end
+
+    (bin/"heartbeat").write <<~EOS
+      #!/bin/sh
+        exec #{libexec}/heartbeat \
+        -path.config #{etc}/heartbeat \
+        -path.home #{prefix} \
+        -path.logs #{var}/log/heartbeat \
+        -path.data #{var}/lib/heartbeat \
+        $@
+    EOS
+  end
+
+  def post_install
+    (var/"lib/heartbeat").mkpath
+    (var/"log/heartbeat").mkpath
+  end
+
+  plist_options :manual => "heartbeat"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN"
+    "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>Program</key>
+        <string>#{opt_bin}/heartbeat</string>
+        <key>RunAtLoad</key>
+        <true/>
+      </dict>
+    </plist>
+  EOS
+  end
+
+  test do
+    require "socket"
+
+    server = TCPServer.new(0)
+    port = server.addr[1]
+    server.close
+
+    (testpath/"config/heartbeat.yml").write <<~EOS
+      heartbeat.monitors:
+      - type: tcp
+        schedule: '@every 5s'
+        hosts: ["localhost:#{port}"]
+        check.send: "hello\\n"
+        check.receive: "goodbye\\n"
+      output.file:
+        path: "#{testpath}/heartbeat"
+        filename: heartbeat
+        codec.format:
+          string: '%{[up]}'
+    EOS
+    pid = fork do
+      exec "#{bin}/heartbeat -path.config #{testpath}/config"
+    end
+    sleep 1
+
+    begin
+      assert_match "hello", pipe_output("nc -c -l #{port}", "goodbye\n")
```

Comment:
is the exit code for `nc` 0? If so you can assert that with a third argument to pipe_output